api/src/main/java/org/pmiops/workbench/api/UserMetricsController.java

```diff
@@ -247,12 +260,21 @@ public boolean hasValidBlobIdIfNotebookNamePresent(DbUserRecentResource dbUserRe
   }
 
   private RecentResource buildRecentResource(
-      ImmutableMap<Long, FirecloudWorkspaceResponse> idToFcWorkspaceResponse,
+      Map<Long, DbWorkspace> idToDbWorkspace,
+      Map<Long, FirecloudWorkspaceResponse> idToFcWorkspaceResponse,
       DbUserRecentResource dbUserRecentResource) {
     RecentResource resource = TO_CLIENT.apply(dbUserRecentResource);
     FirecloudWorkspaceResponse workspaceDetails =
         idToFcWorkspaceResponse.get(dbUserRecentResource.getWorkspaceId());
-    resource.setPermission(workspaceDetails.getAccessLevel());
+    if (workbenchConfigProvider.get().featureFlags.enableBillingLockout) {
+      resource.setWorkspaceBillingStatus(
+          idToDbWorkspace.get(dbUserRecentResource.getWorkspaceId()).getBillingStatus());
```

Comment:
I decided to return this value as part of the RecentResource API model because the response object is used to construct UI resource cards which needs this info. The alternative was to make a getWorkspaces() call for each workspace in the response list and that seemed too heavy handed.

Reply:
Another alternative would be to change `RecentResource` to include the entire workspace, but then I think you're still calling out to firecloud on each request... just not returning as much.

Reply:
yeah... adding the entire workspace feels cleaner but it would defeat the purpose of this change which is to improve performance